packages/devtools_shared/lib/src/memory/memory_json.dart

```diff
@@ -284,25 +284,25 @@ class AllocationMemoryJson extends MemoryJson<ClassHeapStats> {
     assert(oldVersion == 1);
     final updatedPayload = Map<String, dynamic>.of(payload);
     updatedPayload['version'] = version;
-    final oldData = payload['data'];
+    final oldData = payload['data'] as List;
     updatedPayload['data'] = [
-      for (final data in oldData)
+      for (final data in oldData.map((e) => _OldData(e)))
```

Comment:
nit: consider doing the .map operation above where oldData is defined so that we don't have to do this inside the for loop declaration

Reply:
That fits nicely; done.